### PR TITLE
[CHF-201] [CHF-206] Health Check Core Devices

### PR DESCRIPTION
**Pull request by jackchi**
https://smartthings.atlassian.net/wiki/display/CHF/Supported+Health+Check+Devices

Health Checks Core Devices to be compatible with Device-Watch

@ShunmugaSundar - Can you include Parijat / Sushant, I can't find them under members. 
@mortent @workingmonk

### CODE REVIEW

**Comment by jackchi** [comment]
Currently the first time `ping()` tries to construct a Date object from `state.lastActivity`, we get an error:

`7ec9ee27-0432-48c7-acf9-e9cfed0c2220 3:52:55 PM: error groovy.lang.GroovyRuntimeException: Could not find matching constructor for: java.util.Date(java.math.BigInteger) @ line 102`

After checking in the console, the type is `java.Lang.Long`:

> import physicalgraph.device.*
> 
> def d = Device.get("7ec9ee27-0432-48c7-acf9-e9cfed0c2220")
> 
> def t = d.deviceDTO.getDeviceData().lastActivity
> 
> t.getClass()
> 
> ≫ class java.lang.Long 

Not sure why the platform thinks `state.lastActivity` is BigInteger. @Bijnagte might know?

**Comment by workingmonk** [comment]
we will need to regression test all of these devices and make sure none of them throw any errors in live logging/sumo. All the sendEvent messages are before the actual configure and any failure in those messages will leave the devices in the unconfigured state.

**Comment by jackchi** [comment]
Also updated JIRA tickets for QA to use Sumo to search for errors. 

Commit amended and will use Sumo to grep for errors. We are avoiding errors in Date construction now with `(long)` cast and when it's `null` we don't try to construct a Date

**Comment by ShunmugaSundar** [comment]
@jackchi , I am also not able to add Sushant and Parijat, Their write access have been revoked its look like.
Anyways, i have showed them these changes. We have added our review comments

File: devicetypes/smartthings/cree-bulb.src/cree-bulb.groovy [hunk 1]
```diff
@@ -66,6 +67,12 @@ def parse(String description) {
     def resultMap = zigbee.getEvent(description)
     if (resultMap) {
         sendEvent(resultMap)
+        // Temporary fix for the case when Device is OFFLINE and is connected again
+        if (state.lastActivity == null){
+            state.lastActivity = now()
+            sendEvent(name: "deviceWatch-lastActivity", value: state.lastActivity, description: "Last Activity is on ${new Date((long)state.lastActivity)}", displayed: false, isStateChange: true)
```
Comment (ShunmugaSundar) [inline devicetypes/smartthings/cree-bulb.src/cree-bulb.groovy:73]: Why are we typecasting lastActivity as long?
Is it not long by default?
As it stores time in epoch format.

File: devicetypes/smartthings/smartsense-moisture-sensor.src/smartsense-moisture-sensor.groovy [hunk 2]
```diff
@@ -271,6 +278,21 @@ private Map getMoistureResult(value) {
...
 
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+
+	if (state.lastActivity < (now() - (1000 * device.currentValue("checkInterval"))) ){
+		log.info "ping, alive=no, lastActivity=${state.lastActivity}"
+		state.lastActivity = null
+		return zigbee.readAttribute(0x001, 0x0020) // Read the Battery Level
```
Comment (ShunmugaSundar) [inline devicetypes/smartthings/smartsense-moisture-sensor.src/smartsense-moisture-sensor.groovy:289]: After inferring that the device isn't alive, why are we returning zigbee.readAttribute(0x001, 0x0020) i.e. Reading the battery level.
Is it because calling the 'refresh()' is a overhead when we are just trying to update the lastActivity time?

File: devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy [hunk 3]
```diff
@@ -182,7 +182,7 @@ def zwaveEvent(physicalgraph.zwave.Command cmd) {
 
 def configure() {
 	// allow device 10 min to check in; double the periodic reporting interval
-	sendEvent(name: "checkInterval", value: 2*5*60, displayed: false)
+	sendEvent(name: "checkInterval", value: 2*5*60, displayed: false, data: [protocol: "zwave"])
```
Comment (ShunmugaSundar) [inline devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy]: Are we planning to add device category here?

File: devicetypes/smartthings/cree-bulb.src/cree-bulb.groovy (hunk 1 above)
Comment (jackchi) [inline devicetypes/smartthings/cree-bulb.src/cree-bulb.groovy:73]: Yeah, however it ran into situations (first time retrieving state.lastActivity) where the Constructor thinks its a BigInteger
`7ec9ee27-0432-48c7-acf9-e9cfed0c2220 3:52:55 PM: error groovy.lang.GroovyRuntimeException: Could not find matching constructor for: java.util.Date(java.math.BigInteger) @ line 102`

Here's Dylan's answer:

State has to get stored and it is not as a serialized java object it is as json. When it gets retrieved it is parsed back: https://github.com/PhysicalGraph/data-management/blob/master/grails-app/services/physicalgraph/device/cassandra/InstalledDeviceStateService.groovy#L50 The jsonslurper in the old version of groovy we are on doesn’t support  top level primitives so we had to implement our own logic (which largely matches the new behavior of jsonslurper) https://github.com/PhysicalGraph/data-management/blob/master/src/groovy/physicalgraph/util/StringUtils.groovy#L35 it returns a BigInteger

File: devicetypes/smartthings/smartsense-moisture-sensor.src/smartsense-moisture-sensor.groovy (hunk 2 above)
Comment (jackchi) [inline devicetypes/smartthings/smartsense-moisture-sensor.src/smartsense-moisture-sensor.groovy:289]: I'm not certain what the enroll() and config() does. I was just reading an attribute b/c like you said it's light weight.

File: devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy (hunk 3 above)
Comment (jackchi) [inline devicetypes/smartthings/aeon-multisensor.src/aeon-multisensor.groovy]: No plans or need for it yet.
